generator/src/main/java/org/apache/kafka/message/MessageDataGenerator.java

```diff
@@ -1039,6 +1051,9 @@ private void generateFieldHashCode(FieldSpec field) {
         } else if (field.type() instanceof FieldType.Int64FieldType) {
             buffer.printf("hashCode = 31 * hashCode + ((int) (%s >> 32) ^ (int) %s);%n",
                 field.camelCaseName(), field.camelCaseName());
+        } else if (field.type() instanceof FieldType.UUIDFieldType) {
```

Comment:
This is exactly the same as the isStruct / isArray case and can be merged into that clause.